### PR TITLE
catch routing errors

### LINKED ISSUE #22: Better errors
Referencing a controller that doesn't exist in a routes file returns the error

```
TypeError: Cannot call method 'replace' of undefined
```

Better error messages all around would help debugging when using exprestive magic.

### PR DESCRIPTION
**Pull request by charlierudolph**
@alexdavid 

fixes #22
ran into this with the routes issue and thought it would be very helpful

### CODE REVIEW

**Comment by charlierudolph** [comment]
Thinking more about this I'd rather catch these errors on server start. What do you think about just reading the routes files and when adding each route, find the controller and its action? It logs errors if it can't find the controller or action but starts up the server just without the route where the controller are not defined.

**Comment by charlierudolph** [comment]
I retract my previous comment

File: src/controller_initializer.coffee [hunk 1]
```diff
@@ -12,8 +12,13 @@ class ControllerInitializer
...
   # Calls a controller action with args
-  applyAction: ({controllerName, actionName, args}) ->
-    @controllers[camelCase controllerName][actionName] args...
+  applyAction: ({controllerName, actionName, args: [req, res, next]}) ->
+    if not @controllers[camelCase controllerName]
+      next "Missing '#{controllerName}' controller"
+    else if typeof @controllers[camelCase controllerName][actionName] isnt 'function'
+      next "Missing '#{actionName}' action in '#{controllerName}' controller"
+    else
+      @controllers[camelCase controllerName][actionName] req, res, next
```
Comment (alexdavid) [inline src/controller_initializer.coffee]: What do you think about formatting these as guard statements?

``` coffeescript
applyAction: ({controllerName, actionName, args: [req, res, next]}) ->
  controller = @controllers[camelCase controllerName]

  return next "Missing '#{controllerName}' controller" unless controller?

  if typeof controller[actionName] isnt 'function'
    return next "Missing '#{actionName}' action in '#{controllerName}' controller"

  controller[actionName] req, res, next
```

**Comment by alexdavid** [comment]
Thanks for doing this Charlie! :rocket:

**Pull request closed without merging**